smithy-cli/src/main/java/software/amazon/smithy/cli/dependencies/MavenDependencyResolver.java

```diff
@@ -105,13 +108,48 @@ public void addRepository(MavenRepository repository) {
                 addUserInfoAuth(uri, userInfo, builder);
             }
             repository.getHttpCredentials().ifPresent(credentials -> addUserInfoAuth(uri, credentials, builder));
+            addProxyInfo(repository, builder);
             remoteRepositories.add(builder.build());
         } catch (URISyntaxException e) {
             throw new DependencyResolverException("Invalid Maven repository URL: " + repository.getUrl()
                                                   + ": " + e.getMessage());
         }
     }
 
+    private static Proxy getProxy(final String hostPort, final String userPass) {
+        if (hostPort != null) {
```

Comment:
nit - can reverse this if statement to simplify the method by returning null early.